cep-20.2.md

```diff
@@ -0,0 +1,829 @@
+url: url | [url]
+# destination folder in work directory
+target_directory: path
+# rename the downloaded file to this name
+file_name: string
+# hash of the file
+sha256: hex string
+# legacy md5 sum of the file (test both, prefer sha256)
+md5: hex string
+# relative path from recipe file
+patches: [path]
+```
+
+### Local source
+
+A path can be either:
+
+- directory ("../bla")
+- a path to an archive ("../bla.tar.gz")
+- a path to a file ("../bla.txt")
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# if there is a gitignore, adhere to it (or not)
+use_gitignore: bool (defaults to true)
+# destination folder
+target_directory: path
+# rename the downloaded file to this name
+file_name: string
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git: url | path
+# the following 3 keys are mutually exclusive (branch, tag, and rev)
+# branch to checkout to
+branch: string
+# tag to checkout to
+tag: string
+# revision to checkout to (hash or ref)
+rev: string
+# depth of the git clone (mutually exclusive with rev)
+depth: signed integer (defaults to -1 -> not shallow)
+# should this use git-lfs?
+lfs: bool (defaults to false)
+# destination folder in work directory
+target_directory: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [PackageSelector]
+  # dependencies to link against, in the target_platform architecture
+  host: [PackageSelector]
+  # the section below is copied into the index.json and required at package installation
+  run: [PackageSelector]
+  # constrain optional packages (was `run_constrained`)
+  run_constraints: [PackageSelector]
+  # the run exports of this package
+  run_exports: [PackageSelector] OR RunExports
+
+  # the run exports to ignore when calculating the requirements
+  ignore_run_exports:
+    # ignore run exports by name (e.g. `libgcc-ng`)
+    by_name: [string]
+    # ignore run exports that come from the specified packages
+    from_package: [string]
+
+```
+
+#### PackageSelector
+
+A `PackageSelector` in the recipe is currently defined as a string with up to two whitespaces, looking like this:
+
+```
+<name> <version> <build_string>
+
+# examples:
+python
+python 3.8
+python 3.8 h1234567_0
+python >=3.8,<3.9
+python >=3.8,<3.9 h1234567_0
+python 3.9.*
+```
+
+> [!NOTE]
+> `MatchSpec` are defined with many more options in conda. We are sticking to conda-build's definition for the time being.
+> For example, `conda` MatchSpecs allow specifying a channel, a build number (in square brackets) and many additional things.
+> We might homogenize these things later on.
```

Comment:
I think it's fine/good to use a subset for now.
(E.g., in `meta.yaml` you could actually use direct links to package archives or something, but I don't think this is widely supported and will break with `libmamba` solver etc.)

(Also `MatchSpec` is not without issues and we should carefully consider where we use it and whether we might want to replace this with something else generally.)